distributed/shuffle/_scheduler_plugin.py

```diff
@@ -158,15 +166,16 @@ def heartbeat(self, ws: WorkerState, data: dict) -> None:
                 self.heartbeats[shuffle_id][ws.address].update(d)
 
     def get(self, id: ShuffleId, worker: str) -> dict[str, Any]:
-        if exception := self.erred_shuffles.get(id):
-            return {"status": "error", "message": str(exception)}
-        state = self.states[id]
+        if worker not in self.scheduler.workers:
+            raise RuntimeError(f"Scheduler is unaware of this worker {worker!r}")
```

Comment:
I haven't been able to come up with a scenario where this would happen, but given how messy worker shutdown can be, I'm not 100% certain this would _never_ happen. Left it in with a note for now.